### ISSUE TITLE
[suggestion] auto substr if message is too long

### ISSUE THREAD

**Issue by michabbb** [Mizzrym/php_send_nsca#9]
hi there,

it's very hard to debug, when no message gets sent, because you only have some simple `return false` here.
the user usually does not know, that the message has a max length, so i would suggest to update [that part](https://github.com/Mizzrym/php_send_nsca/blob/master/src/SendNsca.php#L123) with:

```php
if (strlen($message)>=512) {
   $message = substr($message,0,511);
}
```
instead of `return false` 😉

**Comment by Mizzrym** [comment]
I think you're right about this one. 
Also, it's time to think about errorhandling and returnvalues in general. 
One could add method to inject a PSR compatible logger and log a warning before just snipping the message, or trigger a warning natively. Same goes for other validations. 
Let me think about that when i have the time to do so. In the meantime, feel free to add more suggestions. 

Thanks.

**Comment by michabbb** [comment]
of course it makes sense, but i would go here step-by-step. the most important thing is "that a message gets sent", without the message, you don´t get any notifications from your nagios, and that is **fatal.**
so when the protocol says, that the max size of the message is 512, you can safely truncate that, because a shorter message is more important that "none message" ;-)

**Comment by Mizzrym** [comment]
You can checkout branch ISSUE-9 for now. 
I'm not quite done testing and i want to add the PSR logger as well, so i didn't want to merge it with master.

**Comment by michabbb** [comment]
Thanks, will test it 👍

**Comment by Mizzrym** [comment]
Fixed in 2.0.0
